src/libAtomVM/memory.h

```diff
@@ -74,17 +211,47 @@ enum MemoryGCResult memory_gc(Context *ctx, int new_size);
  * @param new_heap the destination heap where terms will be copied.
  * @returns a new term that is stored on the new heap.
  */
-term memory_copy_term_tree(term **new_heap, term t, term *mso_list);
+term memory_copy_term_tree(Heap *new_heap, term t);
+
+/**
+ * @brief copies a term to a storage, typically for mailbox messages
+ *
+ * @details deep copies a term to a destination heap, once finished old memory can be freed.
+ * @param storage storage for the copied data, should be large enough
+ * @param heap_end on output, pointer to the end of the term.
+ * @param t term to copy
+ * @returns a boxed term pointer to the new term content that is stored in the storage.
+ */
+term memory_copy_term_tree_to_storage(term *storage, term **heap_end, term t);
 
 /**
- * @brief meakes sure that the given context has given free memory
+ * @brief makes sure that the given context has given free memory
  *
  * @details this function makes sure that at least size terms are available, when not available gc will be performed, any existing term might be invalid after this call.
 
- * @param ctx the target context.
+ * @param heap the target heap.
+ * @param size needed available memory.
+ */
+enum MemoryGCResult memory_ensure_free_maybe_gc(Context *ctx, uint32_t size) MUST_CHECK;
```

Comment:
I understand the idea of changing the intention of the function with the _maybe_gc, but I also feel like this change is gratuitous and just makes a lot of work downstream to fix drivers.  I am not completely opposed to it, if others on the team are okay with it, but in general I would like to try to avoid name changes unless it is really necessary.